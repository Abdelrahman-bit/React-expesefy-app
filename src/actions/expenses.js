import { v4 as uuidv4 } from 'uuid';


export const addExpenses = ({description='', notes='', amount=0, createdAt=0} = {}) => ({
    type: 'ADD_EXPENSE',
    expenses:{
        id: uuidv4(),
        description,
        notes,
        amount,
        createdAt
    }
});

export const editExpense = (id, updateds) => ({
    type: 'EDIT_EXPENSE',
    id,
    updateds

})

export const removeExpense = ({ id } = {}) =>{  
    return{
        type: 'REMOVE_EXPENSE', 
        id
    }
}