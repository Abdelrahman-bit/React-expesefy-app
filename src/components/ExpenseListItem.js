import React from 'react';
import { Link } from 'react-router-dom'


const ExpenseListItem = ({ description, amount, createdAt, id }) =>(
    <div>
        <h3> description: <Link to={`/edit/${id}`}> {description} </Link></h3>
        <p> amount:  {amount} </p>
        <p> created at: {createdAt} </p>
    </div>
);

export default ExpenseListItem;