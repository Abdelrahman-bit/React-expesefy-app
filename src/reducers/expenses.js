

const expensiseReducerDefaultState = [];
export default (state= expensiseReducerDefaultState, action) => {
    switch (action.type) {
        case('ADD_EXPENSE'):
            return [...state, action.expenses]
        case('REMOVE_EXPENSE'):
            return state.filter( (item) => item.id != action.id )
        case('EDIT_EXPENSE'):
            return state.map( (item) => {
                if(item.id == action.id){
                    return {
                        ...item,
                        ...action.updateds
                    }
                }else{
                    return item
                }
            } )
        default:
            return state;
    }
}