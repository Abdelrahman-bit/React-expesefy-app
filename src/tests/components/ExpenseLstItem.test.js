import React from 'react';
import { shallow } from 'enzyme';
import ExpenseListItem from '../../components/ExpenseListItem';
import expenses from '../fixture/expenses';

test('should render the ExpenseListItem', () =>{
    const wapper = shallow( <ExpenseListItem {...expenses[1]} /> );
    expect(wapper).toMatchSnapshot();
})