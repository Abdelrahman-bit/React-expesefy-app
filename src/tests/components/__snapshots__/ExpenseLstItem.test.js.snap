// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the ExpenseListItem 1`] = `
<div>
  <h3>
     description: 
    <Link
      to="/edit/2"
    >
       
      Rent
       
    </Link>
  </h3>
  <p>
     amount:  
    109500
     
  </p>
  <p>
     created at: 
    -345600000
     
  </p>
</div>
`;
